### PR TITLE
Make Xcode report concurrency warnings on the Modules package

### PR DESCRIPTION
**Pull request by crazytonyli**
This PR will introduce about 100+ more warnings. I'll "address" (probably just `@unchecked Sendable`) them in follow up PRs.

As the warnings gone, I'll remove the compiler flags, which means compiler will report errors if new code violates concurrency rules. 

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/303">25.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by kean** [comment]
Apple recommends to work in waves: enable the flags, fix some issues, disable the flags. Continue until there are no more warnings/errors and the flags can be enabled permanently.

**Comment by crazytonyli** [comment]
@kean I thought enabling the concurrency checks (with errors) earlier would be nice, especially considering we have added `@unchecked Sendable` everywhere (in https://github.com/wordpress-mobile/WordPress-iOS/pull/23527) to ignore the concurrency check warnings.

**Comment by kean** [comment]
I'm only suggesting to do it the other way around: fix warnings, then enable the flags in `trunk`.

**Comment by crazytonyli** [comment]
@kean I have removed `.enableUpcomingFeature("StrictConcurrency")`. This PR now should have no impact on the existing code. But future new targets in Modules should be on Swift v6 and get strict concurrency checks.

File: Modules/Package.swift [hunk 1]
```diff
@@ -85,20 +85,22 @@ let package = Package(
...
-        .library(name: "XcodeTarget_WordPressAuthentificator", targets: ["XcodeTarget_WordPressAuthentificator"]),
-        .library(name: "XcodeTarget_WordPressAuthentificatorTests", targets: ["XcodeTarget_WordPressAuthentificatorTests"]),
-        .library(name: "XcodeTarget_ShareExtension", targets: ["XcodeTarget_ShareExtension"]),
-        .library(name: "XcodeTarget_DraftActionExtension", targets: ["XcodeTarget_DraftActionExtension"]),
-        .library(name: "XcodeTarget_NotificationServiceExtension", targets: ["XcodeTarget_NotificationServiceExtension"]),
-        .library(name: "XcodeTarget_Intents", targets: ["XcodeTarget_Intents"]),
-        .library(name: "XcodeTarget_StatsWidget", targets: ["XcodeTarget_StatsWidget"]),
-        .library(name: "XcodeTarget_UITests", targets: ["XcodeTarget_UITests"]),
-    ]
+    static var products: [Product] {
```
Comment (kean) [inline Modules/Package.swift:88]: The previous version seemed a bit more readable.

Comment (crazytonyli) [reply Modules/Package.swift:88]: Swift compiler reports a concurrency check error on the previous version. I can't remember the exact error message, but it's related to the variable being a global variable.

**Review by kean — APPROVED** [review]

**Pull request merged**